### ISSUE TITLE
SSH auth error with googlecompute builder in versions 1.3.0+

### ISSUE THREAD

**Issue by yacn** [hashicorp/packer#9002]
#### Overview of the Issue

When using a packer release later than `1.2.5`, building a google compute / GCP / GCE image fails at the beginning because SSH authentication fails.

We noticed our google compute image builds are failing after we needed to upgrade to packer 1.4.0+ because of the encrypted AMI region copying issue that was fixed in that release. 

I noticed in the `CHANGELOG` that in `1.3.0`, the SSH section was refactored over these two PRs: 

* https://github.com/hashicorp/packer/pull/6613/files
* https://github.com/hashicorp/packer/pull/6621/files

I suspect our issue is related somehow.

I realize that this has been 'broken' for a long time (`1.3.0` was released in 2018) and would think others would've noticed in that time so perhaps the issue is we need to update something in our packer buildfile? I've looked over the settings page for both SSH communicator and GCE builder but nothing sticks out.

#### Reproduction Steps

* build minimal packer buildfile w/ `1.2.5` and watch it successfully authenticate and build
* build minimal packer buildfile w/ `1.3.0` or later and watch it unsuccessfully authenticate via SSH.

### Packer version

Issue exists in packer versions `1.3.0` and beyond. Have tested w/ versions:

* `1.2.3` -- works
* `1.2.4` -- works
* `1.2.5` -- works
* `1.3.0` -- doesn't work
* `1.3.5` -- doesn't work
* `1.4.0` -- doesn't work
* `1.5.5` -- doesn't work

### Simplified Packer Buildfile

It's short but added it to the gist to keep everything together

```json
{
  "variables": {
    "gce_credential_file": "./.chef/morimoto.json",
    "encrypted_data_bag_secret_path": "./.chef/encrypted_data_bag_secret"
  },
  "builders": [
    {
      "type": "googlecompute",
      "account_file": "{{user `gce_credential_file`}}",
      "project_id": "our-project-id",
      "source_image": "ubuntu-1804-bionic-v20191211",
      "machine_type": "n1-standard-4",
      "zone": "us-central1-a",
      "subnetwork": "our-us-central",
      "image_family": "packer-gcp-ssh-issue",
      "image_labels": { "environment": "staging" },
      "image_name": "packer-gcp-ssh-issue-{{timestamp}}",
      "state_timeout": "15m",
      "disk_size": "20",
      "metadata": {
        "application": "issue-repro",
        "nodetype": "issuerepro",
        "environment": "staging"
      },
      "communicator": "ssh",
      "ssh_username": "packer",
      "ssh_agent_auth": true,
      "use_internal_ip": true
    }
  ]
}
```

### Operating system and Environment details

* GCP **`os-login` is _not_ enabled**

Below shouldn't matter but for context:

* Packer builds initiated from inside a jenkins-agent container on GKE scheduled by Jenkins. 
    * [Jenkinsfile being used](https://gist.github.com/yacn/78d99f79a58f1f1bba03f0cfb8f1d449#file-jenkinsfile)

* Container is running Debian 9.9 and is a slightly customized version of the regular jenkins-agent container that jenkins provides.
    * to reproduce issue, was applying updated packer versions to known working version of container, [Dockerfile](https://gist.github.com/yacn/78d99f79a58f1f1bba03f0cfb8f1d449#file-dockerfile-packer-issue-test)

### Log Fragments and crash.log files

* successful build log on packer `1.2.5`
    * [gist link](https://gist.github.com/yacn/78d99f79a58f1f1bba03f0cfb8f1d449/raw/d8bd1f46128cf1ac0b167e9633cb747e4358e1bc/packer-1.2.5-successful.log)
* failed build log on packer `1.3.0`
    * [gist link](https://gist.github.com/yacn/78d99f79a58f1f1bba03f0cfb8f1d449/raw/d8bd1f46128cf1ac0b167e9633cb747e4358e1bc/packer-1.3.0-failed.log)
* failed build log on packer `1.5.5`
    * [gist link](https://gist.github.com/yacn/78d99f79a58f1f1bba03f0cfb8f1d449/raw/d8bd1f46128cf1ac0b167e9633cb747e4358e1bc/packer-1.5.5-failed.log)

I've also looked through the console logs for successful / failed packer runs and don't see any obvious differences like the google_account_daemon crashing or anything.

**Comment by yacn** [comment]
I figured out my own issue:

We were setting `ssh_agent_auth: true` in the packer build files and wrapping our builds in `sshagent` block steps in the `Jenkinsfile` _but_ these both seem to be unnecessary for our environment. I think in the SSH code refactor in 1.3.0, SSH auth precedence in the GCP builder changed to line up w/ others, i.e., `ssh_agent_auth` was now taking precedence as it should've been, and _that_ was the authentication part that was failing, or somehow prior to the refactor it was attempting both SSH agent auth and the generated SSH key auth. Either way, removing that option has solved my issue.

I've tested this by removing `ssh_agent_auth: true` from our packer build file and seeing how we can now build images w/ the latest versions of packer.
